### ISSUE TITLE
ONNX Export error

### ISSUE THREAD

**Issue by PuNeal** [pytorch/pytorch#95867]
### 🐛 Describe the bug

It looks like there is a bug when exporting a pytorch model to onnx format with **torch1.12.0**. I found that the bias is not contained in the Conv node, the structure visualized by Netron is as follows:

![20230302143830](https://user-images.githubusercontent.com/39493003/222350560-9b02f2e2-4506-41d7-8a1b-583cb1f9dcce.png)
The bias of Conv_19 is actually Identity_15. By the way, I tried with **torch1.13.1** but got the same result as torch1.12.0
Then I test with **torch1.9.0**, the result looks like correct as follows:
![微信图片_20230302144414](https://user-images.githubusercontent.com/39493003/222351646-1e5df845-0bbb-42ed-a085-f16712549a42.png)

The code I use to export onnx model is:
```python
import torch
import torchvision
model = torchvision.models.resnet18()
torch.onnx.export(model, torch.randn(1,3,224,224), 'resnet18.onnx', input_names=['input'], output_names=['output'])
```

How can I solve this problem?

### Versions

PyTorch version: 1.13.1+cu117
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.17

Python version: 3.7.16 (default, Jan 17 2023, 22:20:44)  [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.10.16.3-microsoft-standard-WSL2-x86_64-with-debian-buster-sid
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3060
Nvidia driver version: 516.94
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.21.6
[pip3] torch==1.13.1
[pip3] torchaudio==0.9.0
[pip3] torchdynamo==1.13.0
[pip3] torchvision==0.10.0+cu102
[conda] cudatoolkit               10.2.89              hfd86e86_1
[conda] numpy                     1.21.6                   pypi_0    pypi
[conda] torch                     1.13.1                   pypi_0    pypi
[conda] torchaudio                0.9.0                    pypi_0    pypi
[conda] torchdynamo               1.13.0                   pypi_0    pypi
[conda] torchvision               0.10.0+cu102             pypi_0    pypi

**Comment by PuNeal** [comment]
Update.
I made a little modification and the code is as below:
```python
import torch
import torchvision
model = torchvision.models.resnet18(pretrained=True)
torch.onnx.export(model, torch.randn(1,3,224,224), 'resnet18.onnx', input_names=['input'], output_names=['output'])
```
Then I got a correct result.
what is difference between these two case?

**Comment by thiagocrepaldi** [comment]
I would say the untrained model was traced with a bias that was interpreted as a constant instead of a learnable parameter. Therefore, the bias was already contained in the original graph, whereas the pretrained version had a trainable which was added as input to the operator

**Comment by thiagocrepaldi** [comment]
@PuNeal any reason for your concern?

I just repro your issue and as it happens, it is not necessarily a bug, although you might want to not have this behavior :)

In the untrained model, it happens that several conv operators had the same tensor value, and the ONNX export deduplicated them into a single initializer. So there was a single `Conv_194` initializer being used as input by all other `Conv`s in the graph. In the picture below, the left most `Conv` has `Conv_194` being used as a direct input (that was the first occurence of this initializer) while the right `Conv` has an `Identity` op as input which has `Conv_194` as input, creating a `Conv_196` as output. Thus, both `Conv` point to the same initializer. Both `Conv` do have weight+bias inputs. This is the default behavior for onnx export in "eval" model. However, not att is lost; if you want them to point to different values, you can set `training=torch.onnx.TrainingMode.TRAINING` on your `export()` call.

![Image](https://user-images.githubusercontent.com/5469809/225415202-7cac7e18-3219-49a5-a721-b21742bd54bf.png)

In the pretrained model, this doesnt happen because all `Conv` have different initializers values, so deduplication doesnt happen and you see them as you were expecting them


![Image](https://user-images.githubusercontent.com/5469809/225416488-0c5d5cd8-2e27-4729-873f-d572b735012f.png)


All in all, you probably want to use the pretrained model because it should have the most accurate weights, but the untrained graph is also correct

**Comment by thiagocrepaldi** [comment]
Duplicate of https://github.com/pytorch/pytorch/issues/89965

**Comment by PuNeal** [comment]
> @PuNeal any reason for your concern?
> 
> I just repro your issue and as it happens, it is not necessarily a bug, although you might want to not have this behavior :)
> 
> In the untrained model, it happens that several conv operators had the same tensor value, and the ONNX export deduplicated them into a single initializer. So there was a single `Conv_194` initializer being used as input by all other `Conv`s in the graph. In the picture below, the left most `Conv` has `Conv_194` being used as a direct input (that was the first occurence of this initializer) while the right `Conv` has an `Identity` op as input which has `Conv_194` as input, creating a `Conv_196` as output. Thus, both `Conv` point to the same initializer. Both `Conv` do have weight+bias inputs. This is the default behavior for onnx export in "eval" model. However, not att is lost; if you want them to point to different values, you can set `training=torch.onnx.TrainingMode.TRAINING` on your `export()` call.
> 
> ![Image](https://user-images.githubusercontent.com/5469809/225415202-7cac7e18-3219-49a5-a721-b21742bd54bf.png)
> 
> In the pretrained model, this doesnt happen because all `Conv` have different initializers values, so deduplication doesnt happen and you see them as you were expecting them
> 
> ![Image](https://user-images.githubusercontent.com/5469809/225416488-0c5d5cd8-2e27-4729-873f-d572b735012f.png)
> 
> All in all, you probably want to use the pretrained model because it should have the most accurate weights, but the untrained graph is also correct

Hi, how can I do it if I want them to point to different values for onnx export in "eval" model? Thank you!

**Comment by fxmarty** [comment]
@thiagocrepaldi I am wondering why the deduplicate initializer inserts these Identity nodes, while it could be possible just to remove the initializer and point to an other one: https://github.com/onnx/onnx/issues/3278#issuecomment-781948998

**Comment by PuNeal** [comment]
> @PuNeal any reason for your concern?
> 
> I just repro your issue and as it happens, it is not necessarily a bug, although you might want to not have this behavior :)
> 
> In the untrained model, it happens that several conv operators had the same tensor value, and the ONNX export deduplicated them into a single initializer. So there was a single `Conv_194` initializer being used as input by all other `Conv`s in the graph. In the picture below, the left most `Conv` has `Conv_194` being used as a direct input (that was the first occurence of this initializer) while the right `Conv` has an `Identity` op as input which has `Conv_194` as input, creating a `Conv_196` as output. Thus, both `Conv` point to the same initializer. Both `Conv` do have weight+bias inputs. This is the default behavior for onnx export in "eval" model. However, not att is lost; if you want them to point to different values, you can set `training=torch.onnx.TrainingMode.TRAINING` on your `export()` call.
> 
> ![Image](https://user-images.githubusercontent.com/5469809/225415202-7cac7e18-3219-49a5-a721-b21742bd54bf.png)
> 
> In the pretrained model, this doesnt happen because all `Conv` have different initializers values, so deduplication doesnt happen and you see them as you were expecting them
> 
> ![Image](https://user-images.githubusercontent.com/5469809/225416488-0c5d5cd8-2e27-4729-873f-d572b735012f.png)
> 
> All in all, you probably want to use the pretrained model because it should have the most accurate weights, but the untrained graph is also correct

@thiagocrepaldi I met this Error `ValueError Expected more than 1 value per channel when training,`  when setting training=torch.onnx.TrainingMode.TRAINING on my export() call. This is caused by F.batchnorm, how to fix this issue ?
